provider/cloudflare.go

```diff
@@ -270,9 +275,9 @@ func (p *CloudFlareProvider) changesByZone(zones []cloudflare.Zone, changeSet []
 	}
 
 	for _, c := range changeSet {
-		zoneID, _ := zoneNameIDMapper.FindZone(c.ResourceRecordSet.Name)
+		zoneID, _ := zoneNameIDMapper.FindZone(c.ResourceRecordSet[0].Name)
```

Comment:
What if you would have multiple records with different zones pointing to the same endpoint?

Reply:
As with the other case - records for `changeSet` are created internally and should only differ in their `Content`.